tests/libs/__mocks__/prisma.ts

```diff
@@ -1,18 +1,90 @@
+import { PrismockClient } from "prismock";
 import { beforeEach, vi } from "vitest";
-import { mockDeep, mockReset } from "vitest-mock-extended";
 
-import type { PrismaClient } from "@calcom/prisma";
+import logger from "@calcom/lib/logger";
+import * as selects from "@calcom/prisma/selects";
 
 vi.mock("@calcom/prisma", () => ({
   default: prisma,
   prisma,
-  availabilityUserSelect: vi.fn(),
-  userSelect: vi.fn(),
+  ...selects,
 }));
 
+const handlePrismockBugs = () => {
+  const __updateBooking = prismock.booking.update;
+  const __findManyWebhook = prismock.webhook.findMany;
+  const __findManyBooking = prismock.booking.findMany;
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  prismock.booking.update = (...rest: any[]) => {
+    // There is a bug in prismock where it considers `createMany` and `create` itself to have the data directly
+    // In booking flows, we encounter such scenario, so let's fix that here directly till it's fixed in prismock
+    if (rest[0].data.references?.createMany) {
+      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+      // @ts-ignore
+      rest[0].data.references.createMany = rest[0].data.references?.createMany.data;
+      logger.silly("Fixed Prismock bug");
+    }
+    if (rest[0].data.references?.create) {
+      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+      // @ts-ignore
+      rest[0].data.references.create = rest[0].data.references?.create.data;
+      logger.silly("Fixed Prismock bug-1");
+    }
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore
+    return __updateBooking(...rest);
+  };
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  prismock.webhook.findMany = (...rest: any[]) => {
+    // There is some bug in prismock where it can't handle complex where clauses
+    if (rest[0].where?.OR && rest[0].where.AND) {
+      rest[0].where = undefined;
+      logger.silly("Fixed Prismock bug-2");
```

Comment:
Should we keep these?

Reply:
Yeah, I think so. My reasoning is that there might be bug in handling these scenarios and one should be aware of these. Ideally I would want these to be fixed in Prisma, 2 of them are actually not that hard to fix.